libgadget/domain.c

```diff
@@ -471,10 +471,11 @@ domain_check_memory_bound(const int print_details, int64_t *TopLeafWork, int64_t
         ta_free(list_load);
     }
 
-    if(max_load > PartManager->MaxPart)
+    /*Leave a small number of particles for star formation */
+    if(max_load > PartManager->MaxPart * 0.99)
```

Comment:
I think you want to preserve more. Gut feeling is 0.90.

Reply:
That seems a little high - remember that only the star formation mode that forks an extra particle will increase the particle count. So we are talking the SH03 model, not the QuickLya star formation module. A 10% increase in particle count would mean 20% of the gas forming stars in one time step, which seems a lot. How about 0.95?